superduperdb/components/dataset.py

```diff
@@ -60,27 +53,22 @@ def data(self):
     def init(self, db=None):
         """Initialization method."""
         super().init(db=db)
-        self._data = [
-            Document.decode(r, db=self.db).unpack()
-            for r in pickle_decode(self.raw_data)
-        ]
-
-    @override
-    def pre_create(self, db: 'Datalayer') -> None:
-        """Pre-create hook for database operations.
-
-        :param db: The database to use for the operation.
-        """
-        if self.raw_data is None:
+        if self._data is None:
             if self.select is None:
                 raise ValueError('Select cannot be None')
-            data = list(db.execute(self.select))
+            data = list(self.db.execute(self.select))
             if self.sample_size is not None and self.sample_size < len(data):
                 perm = self.random.permutation(len(data)).tolist()
                 data = [data[perm[i]] for i in range(self.sample_size)]
-            self.raw_data = pickle_encode([r.encode() for r in data])
+            self._data = data
 
```

Comment:
I think we don’t need to fix a dataset; instead, the dataset should be the latest data retrieved by the select query.